deploy/kubernetes/images/prow-stable-sidecar-master/image.yaml

```diff
@@ -7,47 +7,46 @@ metadata:
   name: imagetag-csi-provisioner-prow-rc
 imageTag:
   name: k8s.gcr.io/sig-storage/csi-provisioner
-  newTag: "v3.0.0"
+  newTag: "v3.1.0"
 ---
 apiVersion: builtin
 kind: ImageTagTransformer
 metadata:
   name: imagetag-csi-attacher-prow-rc
 imageTag:
   name: k8s.gcr.io/sig-storage/csi-attacher
-  newTag: "v3.2.1"
+  newTag: "v3.4.0"
 ---
 apiVersion: builtin
 kind: ImageTagTransformer
 metadata:
   name: imagetag-csi-resize-prow-rc
 imageTag:
   name: k8s.gcr.io/sig-storage/csi-resizer
-  newTag: "v1.2.0"
+  newTag: "v1.4.0"
 ---
 apiVersion: builtin
 kind: ImageTagTransformer
 metadata:
   name: imagetag-csi-snapshotter-prow-head
 imageTag:
   name: k8s.gcr.io/sig-storage/csi-snapshotter
-  newTag: "v3.0.3"
+  newTag: "v4.0.1"
 ---
 apiVersion: builtin
 kind: ImageTagTransformer
 metadata:
   name: imagetag-csi-node-registrar-prow-rc
 imageTag:
   name: k8s.gcr.io/sig-storage/csi-node-driver-registrar
-  newTag: "v2.3.0"
+  newTag: "v2.5.0"
 ---
 apiVersion: builtin
 kind: ImageTagTransformer
 metadata:
   name: imagetag-csi-gce-driver-prow-rc
 imageTag:
   name: k8s.gcr.io/cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver
```

Comment:
Just to clarify the release candidate validation process:

The `prow-gke-release-staging-rc-master` overlay was used to validate release candidates (say we cut a v1.5.0-rc1 tag, then we apply transformation:

  name: k8s.gcr.io/cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver
  newName: gcr.io/k8s-staging-cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver
  newTag: "v1.5.0-rc1"

Verify the job succeeds and then we cut 1.5.0.
push the image to prod (i.e it will become available in k8s.gcr.io/cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver:1.5.0), then remove the transformation in prow staging overlay and update stable master to use 1.5.0.

So regarding your question whether to keep this transformer for prow-staging-sidecar-master. Yes we should keep the image transformation file.

Reply:
In the meeting we had on overlays, we proposed to rename `prow-gke-release-staging-rc-master` to `prow-stable-sidecar-master`. Would `prow-stable-sidecar-rc-master` be a better name? 

A couple questions on the candidate release process:
We proposed to move `prow-gke-release-staging-rc-master` upstream. This is [the upstream manifest](https://github.com/kubernetes/test-infra/pull/25419/files/892b11ca78e41bc1f497b561e95a9b7e77efa900#r818967770) that I added for prow-gke-release-staging-rc-master. It doesn't test a release candidate currently because it takes the argument release-1-4, but maybe this is wrong and it should not have the release-1-4:
```
--repo=sigs.k8s.io/gcp-compute-persistent-disk-csi-driver=release-1-4
```

Assuming the release-1-4 is incorrect and the argument is: 
```
--repo=sigs.k8s.io/gcp-compute-persistent-disk-csi-driver
```
When we do want to test a release candidate, say v1.5.0-rc1, when we cut a v1.5.0-rc1 tag, we have to 
- Manually add the transformation you described above in this file (deploy/kubernetes/images/prow-stable-sidecar-master/image.yaml) , which applies that newTag: "v1.5.0-rc1". 
- We would NOT need to change the upstream test manifest at all like  "-repo=sigs.k8s.io/gcp-compute-persistent-disk-csi-driver=release-1-5-0-rc1" 

If this is all true then I am thinking I should replace the upstream test manifest for release 1.3 with release 1.4. We don't need both right?

Reply:
stable-master is not supposed to be used to test release candidates. stable-* is the overlay which contains already released candidates. 

the job in upstream can be
ci-gcp-compute-persistent-disk-csi-driver-release-k8s-master-integration which takes the main branch's stable-master overlay. Because we always keep the main branch stable-master poiting to the latest released driver version.

another staging job `ci-gcp-compute-persistent-disk-csi-driver-release-staging-k8s-master-integration` can be added upstream to validate release candidates (e.g 1.5.0-rc1). This job will take the newly created `prow-gke-release-staging-rc-master` overlay.

prow-stable-sidecar-rc-master sounds good to me.

Let's discuss this in a meeting.

Reply:
Yes. I have the prow-gke-release-staging-rc-master (which we renamed to `prow-stable-sidecar-master`)  overlay for the latest the latest rc branch upstream, but thanks for clarifying. I will change it to be just the main branch. Instead of release-1-4. I agree this name is a little confusing still. 

Do these names work? 
prow-gke-release-staging-rc-master→prow-stable-sidecar-latest-rc
rename prow-gke-release-staging-head→prow-canary-sidecar

Reply:
yeah let's meet.

I think the overriding goal is to simplify & reduce duplication between internal & upstream